ts/Core/Axis/Axis.ts

```diff
@@ -4427,8 +4427,8 @@ namespace Axis {
         dataMin: number;
         max: number;
         min: number;
-        userMax: number;
-        userMin: number;
+        userMax: number | undefined;
+        userMin: number | undefined;
```

Comment:
That type annotation actually has a slightly different meaning (see [here](https://www.typescriptlang.org/docs/handbook/2/objects.html#optional-properties)): 
- `userMax?: number;` would imply that `userMax` may or may not be present/provided in the `ExtremesObject`, but when present must be of type `number`.
- `userMax: number | undefined;` would imply that `userMax` will always be present in the object, and its value will either be of type `number` or `undefined`. 

The latter seems like the correct typing to me, since there shouldn't be a circumstance where it would make sense to allow `getExtremes()` to return an object that omits `userMax` or `userMin`, and we Highcharts does have situations where it returns an object where `userMax` or `userMin` can have an `undefined` value. In the current code, it always does provide them, returning either undefined or number: https://github.com/highcharts/highcharts/blob/419f1d9358e553666fbb3e8297a33d46ac32afb0/ts/Core/Axis/Axis.ts#L2783-L2794

(Side note: I realized we no longer need the `as any` assertion there, so I've updated that.)

Let me know if you agree with the above reasoning. Thanks!

Reply:
Optional properties include `undefined` as a valid value. The difference is that with `?` the property key may not be defined on the object.

Reply:
Right, but in this case, I am saying that it makes more sense to specify the type as saying the value is always present, but that it may be undefined, rather than saying that it may not be present in the boject. As seen in the above code, `userMin` and `userMax` _are_ always part of the returned object for an `ExtremesObject`. Why relax the typing to say that they can be omitted? It seems like you should always expect `getExtremes` to return an object consisting of all fields, not sometimes have some values missing.